sympy/ntheory/residue_ntheory.py

```diff
@@ -742,6 +744,51 @@ def _nthroot_mod1(s, q, p, all_roots):
         return res
     return min(res)
 
+def _nthroot_mod_composite(a, n, m):
+    """
+    Find the solutions to ``x**n = a mod m`` when m is not prime.
+    """
+    f = factorint(m)
+    dd = {}
+    for p, e in f.items():
+        tot_roots = set()
+        if e == 1:
+            root = nthroot_mod(a, n, p, True) or []
+            if (pow(p, n) - a) % p == 0:
```

Comment:
`nthroot_mod` should return also this case.

Reply:
No is doesn't. `nthroot_mod(0, 5, 17, True)` returns None but it should return [0].  see issue #18212

Reply:
So let's get the PR that fixes that in first.

Reply:
i am working on it. Will send it by tomorrow.

Reply:
I think `is_nthpow_residue(a, n, m)` works only when m is prime. Because `is_nthpow_residue(36010, 8, 87382)` returns false but it has solutions `[40208, 47174]`

Reply:
`nthroot_mod(0, 5, 17, True)` returns [0] in commit 8ac5ad200d6929d293afc5fd59b412d14b6f7edf.
in which commit, does it return `None`?